tests/e2e/actions.go

```diff
@@ -268,72 +391,220 @@ type SubmitConsumerAdditionProposalAction struct {
+		ChainId:                  chainID,
+		Metadata:                 metadata,
+		InitializationParameters: initParams,
+		PowerShapingParameters:   powerShapingParams,
+	}
+
+	content, err := json.Marshal(rec)
+	if err != nil {
+		log.Fatalf("failed marshalling MsgCreateConsumer: %s", err.Error())
+	}
+	jsonFile := "/create_consumer.json"
+	bz, err := tr.target.ExecCommand(
+		"/bin/bash", "-c", fmt.Sprintf(`echo '%s' > %s`, content, jsonFile),
+	).CombinedOutput()
+	if err != nil {
+		log.Fatal(err, "\n", string(bz))
+	}
+
+	// Send consumer chain creation
+	cmd := tr.target.ExecCommand(
+		tr.testConfig.chainConfigs[providerChain].BinaryName,
+		"tx", "provider", "create-consumer", jsonFile,
+		`--from`, `validator`+fmt.Sprint(validator),
+		`--chain-id`, string(tr.testConfig.chainConfigs[providerChain].ChainId),
+		`--home`, tr.getValidatorHome(providerChain, validator),
+		`--gas`, `900000`,
+		`--node`, tr.getValidatorNode(providerChain, validator),
+		`--keyring-backend`, `test`,
+		"--output", "json",
+		`-y`,
+	)
+
+	bz, err = cmd.CombinedOutput()
+	if err != nil {
+		log.Fatal("create consumer failed ", "error: ", err, "output: ", string(bz))
+	}
+
+	txResponse := &TxResponse{}
+	err = json.Unmarshal(bz, txResponse)
+	if err != nil {
+		log.Fatalf("unmarshalling tx response on create-consumer: %s, json: %s", err.Error(), string(bz))
+	}
+
+	if txResponse.Code != 0 {
+		log.Fatalf("sending transaction failed with error code %d, Log:'%s'", txResponse.Code, txResponse.RawLog)
+	}
+
+	// TODO: introduce waitForTx
+	tr.waitBlocks(providerChain, 2, 10*time.Second)
+
+	// Get Consumer ID from transaction
+	cmd = tr.target.ExecCommand(
+		tr.testConfig.chainConfigs[providerChain].BinaryName,
+		"query", "tx", txResponse.TxHash,
+		`--node`, tr.getValidatorNode(providerChain, validator),
+		"--output", "json",
+	)
+	bz, err = cmd.CombinedOutput()
+	if err != nil {
+		log.Fatal("not able to query tx containing creation-consumer: cmd:", cmd, "err:", err.Error(), "out:", string(bz))
+	}
+
+	err = json.Unmarshal(bz, txResponse)
+	if err != nil {
+		log.Fatalf("unmarshalling tx containing create-consumer: %s, json: %s", err.Error(), string(bz))
+	}
+
+	consumerId := ""
+	for _, event := range txResponse.Events {
+		if event.Type != "consumer_creation" {
+			continue
+		}
+		attr, exists := event.GetAttribute("consumer_id")
+		if !exists {
+			log.Fatalf("no event with consumer_id found in tx content of create-consumer: %v", event)
+		}
+		consumerId = attr.Value
+	}
+	if consumerId == "" {
+		log.Fatalf("no consumer-id found in consumer creation transaction events for chain '%s'. events: %v", consumerChain, txResponse.Events)
+	}
+
+	cfg, exists := tr.testConfig.chainConfigs[e2e.ChainID(chainID)]
+	if !exists {
+		log.Fatal("no chain config found for consumer chain", chainID)
+	}
+	if cfg.ConsumerId != "" && cfg.ConsumerId != e2e.ConsumerID(consumerId) {
+		log.Fatal("chain ", chainID, " registered already with a different consumer ID", consumerId)
```

Comment:
Space at the end of `"chain "` while not at the end of the second string. Different style with what you had in previous `Fatalf` where you use `:` at the end of a name.